### PR TITLE
Rename Azure modules:  facts -> info

### PR DESCRIPTION
**Pull request by smile37773**
##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
azure_rm_aks_info
azure_rm_aksversion_info
azure_rm_applicationsecuritygroup_info
azure_rm_appserviceplan_info
azure_rm_availabilityset_info
azure_rm_autoscale_info

### CODE REVIEW

File: lib/ansible/config/module_defaults.yml [hunk 1]
```diff
@@ -522,15 +522,15 @@ groupings:
   - azure
   azure_rm_aks:
   - azure
-  azure_rm_aks_facts:
+  azure_rm_aks_info:
```
Comment (s-hertel) [inline lib/ansible/config/module_defaults.yml]: Please also retain the previous name since module_defaults doesn't set aliases.

**Review by s-hertel — CHANGES_REQUESTED** [review]

File: lib/ansible/config/module_defaults.yml (hunk 1 above)
Comment (smile37773) [reply lib/ansible/config/module_defaults.yml]: ok, thanks.

Comment (s-hertel) [reply lib/ansible/config/module_defaults.yml]: I'd still add the _info modules to the list. Just keep the _facts name alternatives there too.

File: lib/ansible/modules/cloud/azure/azure_rm_aks_info.py [hunk 2]
```diff
@@ -14,9 +14,9 @@
 
 DOCUMENTATION = '''
 ---
-module: azure_rm_aks_facts
+module: azure_rm_aks_info
 
-version_added: "2.6"
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_aks_info.py:19]: just one note here, version_added should stay the same as in old facts module

**Comment by zikalino** [comment]
we should also display depreciation info, code should look something like this in every module:

    if module._name == '<module name>_facts':
        module.deprecate("The '<module name>_facts' module has been renamed to '<module name>_info'", version='2.13')

File: lib/ansible/modules/cloud/azure/azure_rm_aks_info.py (hunk 2 above)
Comment (smile37773) [reply lib/ansible/modules/cloud/azure/azure_rm_aks_info.py:19]: if i don't change the version_added, it cannot pass the check

Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_aks_info.py:19]: ok

Comment (felixfontein) [reply lib/ansible/modules/cloud/azure/azure_rm_aks_info.py:19]: I don't think bumping the version is a good idea. In all cases I renamed a module, I never had to do this.

File: lib/ansible/config/module_defaults.yml (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/config/module_defaults.yml]: The `_info` modules are now missing. Will they be added to `module_defaults.yml` in another PR?

**Comment by felixfontein** [comment]
Please see https://github.com/ansible/ansible/commit/543801319180d606bbe475a3c7c42a73ecbb04c0#r34856104

**Comment by smile37773** [comment]
> Please see [5438013#r34856104](https://github.com/ansible/ansible/commit/543801319180d606bbe475a3c7c42a73ecbb04c0#r34856104)

Thank you for your suggestion. At first, i try to use self.module but it failed the checks and told me there is no attribute "module" i feel really strange so i have to use AnsibleModule()

**Comment by felixfontein** [comment]
I tested whether instantiating `AnsibleModule` twice is a problem. It seems that it works fine. (There is some code in `module_utils/basic.py` which makes sure that the module input is not read twice from stdin if `AnsibleModule` is instantiated multiple times.)

Anyway, `self.module` is initialized in https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/azure_rm_common.py#L288, i.e. it will be available **after** the `super(AzureRMAKSVersion, self).__init__()` call. Especially because the first `AnsibleModule` instance is thrown away, with the result that the warning you pass to it will never be returned!

**Comment by smile37773** [comment]
> I tested whether instantiating `AnsibleModule` twice is a problem. It seems that it works fine. (There is some code in `module_utils/basic.py` which makes sure that the module input is not read twice from stdin if `AnsibleModule` is instantiated multiple times.)
> 
> Anyway, `self.module` is initialized in https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/azure_rm_common.py#L288, i.e. it will be available **after** the `super(AzureRMAKSVersion, self).__init__()` call. Especially because the first `AnsibleModule` instance is thrown away, with the result that the warning you pass to it will never be returned!

Thank you so much!

**Pull request merged**